CefSharp/DependencyChecker.cs

```diff
@@ -0,0 +1,196 @@
+using System.IO;
+using System.Reflection;
+using System.Text;
+
+namespace CefSharp
+{
+    /// <summary>
+    /// DependencyChecker provides a known list of Cef/CefSharp dependencies and 
+    /// provides helper methods to check for their existance.
+    /// </summary>
+    public static class DependencyChecker
+    {
+        public const string LocalesPackFile = @"locales\en-US.pak";
+
+        /// <summary>
+        /// List of Cef Dependencies
+        /// </summary>
+        public static string[] CefDependencies =
+        {
+            // CEF core library
+            "libcef.dll",
+            // Unicode support
+            "icudtl.dat"
+        };
+
+        /// <summary>
+        /// List of Cef Resources (pack files)
+        /// </summary>
+        public static string[] CefResources =
+        {
+            // Pack Files
+            // Note: Contains WebKit image and inspector resources.
+            "devtools_resources.pak",
+            "cef.pak",
+            "cef_100_percent.pak",
+            "cef_200_percent.pak"
+        };
+
+        public static string[] CefOptionalDependencies =
+        {
+            // Angle and Direct3D support
+            // Note: Without these components HTML5 accelerated content like 2D canvas, 3D CSS and WebGL will not function.
+            "libEGL.dll",
+            "libGLESv2.dll",
+            "d3dcompiler_43.dll",
+            "d3dcompiler_47.dll",
+            // PDF support
+            // Note: Without this component printing will not function.
+            "pdf.dll",
+            //FFmpeg audio and video support
+            // Note: Without this component HTML5 audio and video will not function.
+            "ffmpegsumo.dll"
+        };
+
+        /// <summary>
+        /// List of CefSharp Dependencies
+        /// </summary>
+        public static string[] CefSharpDependencies =
+        {
+            "CefSharp.Core.dll",
+            "CefSharp.dll",
+            "CefSharp.BrowserSubprocess.Core.dll",
+            "CefSharp.BrowserSubprocess.exe"
+        };
+
+        /// <summary>
+        /// CheckDependencies iterates through the list of Cef and CefSharp dependencines
+        /// relative to the path provided and returns a list of missing ones
+        /// </summary>
+        /// <param name="checkOptional">check to see if optional dependencies are present</param>
+        /// <param name="packLoadingDisabled">Is loading of pack files disabled?</param>
+        /// <param name="path">path to check for dependencies</param>
+        /// <param name="resourcesDirPath"></param>
+        /// <param name="localePackFile">The locale pack file e.g. <see cref="LocalesPackFile"/> </param>
+        /// <returns>List of missing dependencies, if all present an empty List will be returned</returns>
+        public static List<string> CheckDependencies(bool checkOptional, bool packLoadingDisabled, string path, string resourcesDirPath, string localePackFile = LocalesPackFile)
```

Comment:
Whenever I see gobs of bool parameters I usually think: Are you really sure using a flag enum isn't the correct thing to do here?